docs/experiments/extract-mode.md

```diff
@@ -0,0 +1,103 @@
+---
+id: extract-mode
+append_help_link: true
+description: "This article explains the extract mode, which allows for easier handling of files containing more than one language."
+---
+
+# Extract mode
+
+## Introduction
+
+Extract mode enables you to run existing rules on subsections of files where the rule language is different than the language of the file. For example, running a Javascript rule on code contained inside of script tags in an HTML document.
+
+## Example of extract mode
+
+Without extract mode, writing rules to validate template, Markdown or configuration files which contain code in another language can be burdensome and require significant rule duplication.
+
+Let's take the following Bash rule as an example (a simplified version of [our `curl-eval` rule](https://github.com/returntocorp/semgrep-rules/blob/release/bash/curl/security/curl-eval.yaml)):
```

Comment:
The general form of "<rule id> rule" as a way to refer to a specific rule feels very awkward to me. Do we have a standard convention here? If we'd like to avoid possessive determiners, how about "`curl-eval`, from the Semgrep Registry"?

Reply:
[migrated to other branch]